src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities.cs

```diff
@@ -62,6 +67,46 @@ public static KeyValueLogScope CreateArtifactScope(string artifactType, IEndpoin
             return scope;
         }
 
+        public static string GetLogsContentType(LogFormat format)
+        {
+            if (format == LogFormat.EventStream)
+            {
+                return ContentTypes.TextEventStream;
+            }
+            else if (format == LogFormat.NDJson)
+            {
+                return ContentTypes.ApplicationNdJson;
+            }
+            else if (format == LogFormat.JsonSequence)
+            {
+                return ContentTypes.ApplicationJsonSequence;
+            }
+            else
+            {
+                return ContentTypes.TextEventStream;
+            }
+        }
+
+        public static async Task StartLogsPipeline(TaskCompletionSource<object> startCompletionSource, LogFormat format, IEndpointInfo endpointInfo, EventLogsPipelineSettings settings, Stream outputStream, CancellationToken token)
+        {
+            using var loggerFactory = new LoggerFactory();
+
+            loggerFactory.AddProvider(new StreamingLoggerProvider(outputStream, format, logLevel: null));
+
+            var client = new DiagnosticsClient(endpointInfo.Endpoint);
+
+            await using EventLogsPipeline pipeline = new EventLogsPipeline(client, settings, loggerFactory);
+
+            Task pipelineTask = pipeline.RunAsync(token);
+
+            if (null != startCompletionSource)
```

Comment:
You need to await the start of the pipeline before signaling that it has started:

```cs
Task runTask = pipeline.RunAsync(token);

await pipeline.SessionStarted.WithCancellation(token);

if (null != startCompletionSource)
```

Reply:
It seemed that I couldn't await `SessionStarted` on pipeline, so I followed a precedent I saw in another file for this (I also couldn't seem to put WithCancellation on it). Just make sure you double check that my interpretation is what you intended.

Reply:
I forgot that the property is implemented explicitly: https://github.com/dotnet/diagnostics/blob/9ccbd1e94c1571c4af35bda26e580474aed3a37f/src/Microsoft.Diagnostics.Monitoring.EventPipe/EventSourcePipeline.cs#L19

So what you did here is the correct solution to access it.